pkg/condition/conditions.go

```diff
@@ -106,6 +118,35 @@ func (c *condition) IsWindowsIPAMEnabled() bool {
 	return false
 }
 
+func (c *condition) IsWindowsPrefixDelegationEnabled() bool {
+	if c.IsOldVPCControllerDeploymentPresent() {
+		return false
+	}
+
+	// Return false if configmap not present/any errors
+	vpcCniConfigMap, err := c.K8sAPI.GetConfigMap(config.VpcCniConfigMapName, config.KubeSystemNamespace)
+
+	if err == nil && vpcCniConfigMap.Data != nil {
```

Comment:
From design point of view, if users turn on Windows prefix, that should indicate the user is using Windows. I wonder if it is necessary to have two toggles for users:
- EnableWindowsIPAMKey
- EnableWindowsPrefixDelegationKey

Will that work same if `EnableWindowsPrefixDelegationKey` is true, bypass `EnableWindowsIPAMKey` and turn on Windows support?

Reply:
You are right. We can make `EnableWindowsPrefixDelegationKey` bypass `EnableWindowsIPAMKey` here. 

If users need to turn off PD but still keep Windows support, they will have to explicitly turn on `EnableWindowsIPAMKey` though.

Reply:
@haouc Just wanted to clarify: what happens if users set conflicting flags, e.g. PD=true, IPAM=false. In this case, which flag should override? If following the above logic of PD flag overriding IPAM flag, does it mean IPAM is still enabled even though it is explicitly set to false?

Reply:
Yeah, my opinion is if PD is on Windows should be considered as on. This can be called out explicitly in doc that the best practice is turn on IPAM and / or turn on PD. If users only turn on PD, the controller will by default think the user turns on IPAM. 
If you believe requiring 1, turn on IPAM, 2 if wanted, turn on PD. If only turning on PD, this can be considered as a misconfiguration. This should also work to me. 
It is basically UX thing.

Reply:
After getting input from Product, we will keep these two flags as independent for now, and no override behavior here to avoid future confusion. PD flag will only be effective when IPAM is enabled.